src/result_object/spec.js

```diff
@@ -96,7 +108,54 @@ describe('Test PassableResponse class', () => {
         });
     });
 
-    describe('Test getErrors method', () => {
+    describe('method: runCompletionCallbacks', () => {
+        it('Should run all functions in `completionCallbacks` list', () => {
+            const res = new ResultObject(faker.lorem.word());
+            const fn1 = sinon.spy();
+            const fn2 = sinon.spy();
+            const fn3 = sinon.spy();
+            res.done(fn1).done(fn2).done(fn3);
+            res.runCompletionCallbacks();
+            expect(fn1.calledOnce).to.equal(true);
+            expect(fn2.calledOnce).to.equal(true);
+            expect(fn3.calledOnce).to.equal(true);
+        });
+
+        it('Should pass current ResultObject instance to the callback', () => {
+            const res = new ResultObject(faker.lorem.word());
+            res.done((instance) => {
+                expect(res).to.equal(instance);
+            });
+            res.runCompletionCallbacks();
+        });
+    });
+
+    describe('method: done', () => {
+        it('Should add given callback to `completionCallbackList`', () => {
+            const res = new ResultObject(faker.lorem.word());
+            expect(res.completionCallbacks).to.have.lengthOf(0);
+            res.done(noop);
+            expect(res.completionCallbacks).to.have.lengthOf(1);
+            res.completionCallbacks[0] === noop;
+        });
+
+        it('Should return without adding non-function values', () => {
+            const res = new ResultObject(faker.lorem.word());
+            [0, 1, null, undefined, new Promise(noop), false, true, [], {}].forEach((v) => res.done(v));
+            expect(res.completionCallbacks).to.have.lengthOf(0);
+        });
+
+        it('Should push given callback as last element in the array', (done) => {
+            const res = new ResultObject(faker.lorem.word());
+
+            res.done(noop);
+            res.done(() => done());
+
+            res.completionCallbacks[1]();
+        });
```

Comment:
I think `const res = new ResultObject(faker.lorem.word());` should be place in `beforeEach`